extension/jaegerremotesampling/internal/remote_strategy_store.go

```diff
@@ -27,15 +31,30 @@ type grpcRemoteStrategyStore struct {
 func NewRemoteStrategyStore(
 	conn *grpc.ClientConn,
 	grpcClientSettings *configgrpc.GRPCClientSettings,
-) strategystore.StrategyStore {
+	reloadInterval time.Duration,
+) (strategystore.StrategyStore, io.Closer) {
+	cache := newNoopStrategyCache()
+	if reloadInterval > 0 {
+		cache = newServiceStrategyCache(reloadInterval)
+	}
+
 	return &grpcRemoteStrategyStore{
 		headerAdditions: grpcClientSettings.Headers,
 		delegate:        grpcstore.NewConfigManager(conn),
-	}
+		cache:           cache,
+	}, cache
 }
 
 func (g *grpcRemoteStrategyStore) GetSamplingStrategy(ctx context.Context, serviceName string) (*sampling.SamplingStrategyResponse, error) {
-	return g.delegate.GetSamplingStrategy(g.enhanceContext(ctx), serviceName)
+	if cachedResponse, ok := g.cache.get(serviceName); ok {
+		return cachedResponse, nil
+	}
+	freshResult, err := g.delegate.GetSamplingStrategy(g.enhanceContext(ctx), serviceName)
+	if err != nil {
+		return nil, fmt.Errorf("GetSamplingStrategy call to remote failed: %w", err)
```

Comment:
An idiomatic go error keeps the error message text as lower case to ensure that when it is logged like
`fmt.Println("An error happened:", err)` that it reads like a normal sentence.

I suggest renaming to something like `remote call failed: %w`